Assets/Scripts/Possession/PossessionController.cs

```diff
@@ -0,0 +1,53 @@
+using UnityEngine;
+
+public class PossessionController : MonoBehaviour
+{
+    private IPossessable currentPossession;
+    private Camera mainCamera;
+
+    private void Start()
+    {
+        mainCamera = Camera.main;
+        currentPossession = GetComponent<IPossessable>();
+    }
+
+    private void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.E))
+        {
+            Possess();
+        }
+    }
+
+    /// <summary>
+    /// If camera is looking at possessable object possess it and unpossess current possession
+    /// </summary>
+    private void Possess()
+    {
+        IPossessable possessable = LookForPossessableObject();
+        if (possessable == null)
+        {
+            return;
+        }
+        currentPossession.Unpossess();
+        possessable.Possess();
+        currentPossession = possessable;
+    }
+
+    /// <summary>
+    /// Checks if the mainCamera is pointing at an object with the IPossessable interface
+    /// </summary>
+    /// <returns>The found IPossessable interface or null when no IPossessable interface is found</returns>
+    private IPossessable LookForPossessableObject()
+    {
+        RaycastHit hit;
+        if (Physics.Raycast(mainCamera.transform.position, mainCamera.transform.forward, out hit, Mathf.Infinity))
```

Comment:
Instead of Infinite recast a variable for the possession range would be better

Reply:
I added a max range variable